### PR TITLE
Support webpack

### PR DESCRIPTION
**Pull request by amowu**
When I use node-wit in my APEX webpack project (https://github.com/apex/apex/tree/master/_examples/babel-webpack),
I can't import or require this, because request package has problem,
so I replace it by axios.

### CODE REVIEW

**Comment by blandinw** [comment]
@amowu Thanks for the contribution!
Making this available in both Node.js and browser is a great idea.
Implementation-wise, we'd like to use something more standard, like the `fetch` API.

It has several advantages:
- it's standard
- browsers don't need to download any extra code
- `fetch` as an API has popular polyfills for older browsers / node.js (https://github.com/github/fetch, https://github.com/bitinn/node-fetch)

If you can amend your PR to use `fetch` (and make sure if works with `node-fetch`) instead, we'll be happy to merge that in!

**Comment by DiegoRBaquero** [comment]
node-wit should work with browserify without any extra change

**Comment by blandinw** [comment]
@DiegoRBaquero yes, browserify will import [http-browserify](https://github.com/substack/http-browserify) by default. `request` then becomes a 2MB dependency (cf. https://github.com/request/request/issues/2090) that the browser needs to download.
We'd gladly accept a PR to use `fetch` instead.

**Comment by DiegoRBaquero** [comment]
@blandinw Interesting issue, thanks for the share.

**Comment by amowu** [comment]
Hi @blandinw I already updated this PR from axios to node-fetch!

File: lib/wit.js [hunk 1]
```diff
@@ -101,39 +103,42 @@ const clone = (obj) => {
...
   };
 
   this.converse = (sessionId, message, context, cb) => {
-    const options = {
-      uri: '/converse',
-      method: 'POST',
-      qs: { 'session_id': sessionId },
-      json: context,
-    };
+    let qs = 'session_id=' + sessionId;
```
Comment (blandinw) [inline lib/wit.js:150]: can we use `encodeURIComponent` on `sessionId` here?

File: lib/wit.js [hunk 2]
```diff
@@ -101,39 +103,42 @@ const clone = (obj) => {
...
-  this.req = request.defaults({
-    baseUrl: process.env.WIT_URL || 'https://api.wit.ai',
-    strictSSL: false,
-    json: true,
-    headers: {
-      'Authorization': 'Bearer ' + token,
-    },
-  });
+  const baseURL = process.env.WIT_URL || 'https://api.wit.ai';
+  const body = true;
```
Comment (blandinw) [inline lib/wit.js]: why are we setting body to `true` here?

File: lib/wit.js [hunk 3]
```diff
@@ -101,39 +103,42 @@ const clone = (obj) => {
...
   this.message = (message, cb) => {
-    const options = {
-      uri: '/message',
+    fetch(baseURL + '/message', {
       method: 'GET',
-      qs: { q: message },
-    };
-    this.req(options, makeWitResponseHandler('message', l, cb));
+      headers: headers,
+      body: body,
```
Comment (blandinw) [inline lib/wit.js]: why send the `body` in the call to `/message` ?

File: lib/wit.js [hunk 4]
```diff
@@ -101,39 +103,42 @@ const clone = (obj) => {
...
+    }).then(response => Promise.all([response.json(), response.status]))
+      .then(makeWitResponseHandler('message', l, cb))
```
Comment (blandinw) [inline lib/wit.js]: let's put this in a `const` above:

```
const handler = makeWit...
```

File: lib/wit.js [hunk 5]
```diff
@@ -101,39 +103,42 @@ const clone = (obj) => {
...
-      options.qs.q = message;
+      qs += '&q=' + encodeURIComponent(message);
     }
-    this.req(options, makeWitResponseHandler('converse', l, cb));
+    fetch(baseURL + '/converse?' + qs, {
+      method: 'POST',
+      headers: headers,
+      body: JSON.stringify(context),
+    }).then(response => Promise.all([response.json(), response.status]))
+      .then(makeWitResponseHandler('converse', l, cb))
```
Comment (blandinw) [inline lib/wit.js]: let's put this in a `const` above:

```
const handler = makeWit...
```

**Comment by blandinw** [comment]
@amowu Thank you for the contribution!
I added a few comments. There's also a conflict in lib/wit.js with master when rebasing.
Let me know if you'd like me to fix those and merge myself. In that case, I'm not sure if you'll still have your name on the commit (from https://help.github.com/articles/checking-out-pull-requests-locally/).

**Comment by blandinw** [comment]
@amowu I implemented the fixes in this PR: https://github.com/amowu/node-wit/pull/1/commits
Please merge and update your PR and we're good 😊

File: lib/wit.js (hunk 3 above)
Comment (blandinw) [inline lib/wit.js]: also forgot to actually send the `message`

**Comment by amowu** [comment]
@blandinw done! and thank you so much 😄

**Comment by blandinw** [comment]
🎉

**Pull request merged**